python/pyarrow/src/arrow/python/arrow_to_pandas.cc

```diff
@@ -2320,6 +2329,17 @@ class ConsolidatedBlockCreator : public PandasBlockCreator {
       if (arrays_[column_index]->type()->id() == Type::EXTENSION) {
         arrays_[column_index] = GetStorageChunkedArray(arrays_[column_index]);
       }
+      // In case of a RunEndEncodedArray default to the storage type
+      else if (arrays_[column_index]->type()->id() == Type::RUN_END_ENCODED) {
+        ARROW_ASSIGN_OR_RAISE(Datum decoded,
+                              compute::RunEndDecode(arrays_[column_index]));
```

Comment:
Can you move this part into the `GetDecodedChunkedArray` helper? (so it is fully equivalent like the `GetStorageChunkedArray` one? Or the problem is that in this case it can fail and we need to handle that?)

Reply:
>  Or the problem is that in this case it can fail and we need to handle that?

Yes, exactly. I thought it was easier to just move it out. If `GetDecodedChunkedArray` stays minimal I thought we might not even use the helper function at all.

Reply:
I think you should be able to write it such that `GetDecodedChunkedArray` returns a Result, and then still use that here. For example, it should be possible to do something like `ARROW_ASSIGN_OR_RAISE(arrays_[column_index], GetDecodedChunkedArray(arrays_[column_index]))`, I think

Reply:
Done: https://github.com/apache/arrow/pull/40661/commits/6308551254279f4387da4c0ceb6bf8b3edc6de90